Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp

```diff
@@ -146,15 +151,9 @@ static int createFlatpakInfo()
     static size_t size;
 
     if (!data.get()) {
-        // xdg-desktop-portal relates your name to certain permissions so we want
-        // them to be application unique which is best done via GApplication.
-        GUniqueOutPtr<GError> error;
-        const char* appID = applicationId(&error.outPtr());
-        if (!appID)
-            g_error("Unable to configure xdg-desktop-portal access in the WebKit sandbox: %s", error->message);
-
+        String appID = applicationId();
         GUniquePtr<GKeyFile> keyFile(g_key_file_new());
-        g_key_file_set_string(keyFile.get(), "Application", "name", appID);
+        g_key_file_set_string(keyFile.get(), "Application", "name", appID.utf8().data());
```

Comment:
/* Unofficial Review */
appID.utf8().data() -> ensureApplicationID().utf8().data(). Then, we can remove the declaration of appID above.

By the way, the only user of the applicationID() is here. char* -> String -> String (copy) ->  char* is inefficient. I suggest keeping const char* instead of String for the return value.

Reply:
The API is no longer `const char*` because it dynamically generates a string, so a copy will always be around. `String` vs `GUniquePtr<char>` isn't a big difference IMO.